user_tools/src/spark_rapids_pytools/rapids/rapids_tool.py

```diff
@@ -473,13 +474,15 @@ def cache_all_dependencies(dep_arr: List[dict]):
                     futures = executor.submit(cache_single_dependency, dep)
                     futures.add_done_callback(exception_handler)
                     futures_list.append(futures)
-                for future in futures_list:
-                    try:
-                        result = future.result(timeout=360)
+                try:
+                    # TODO: Note that we do not set timeout here to avoid failing when network
+                    #       is slow downloading one of the resources.
+                    for future in concurrent.futures.as_completed(futures_list):
+                        result = future.result()
```

Comment:
Should we try increasing the timeout rather than removing it altogether?  I am trying to understand if the CI job runs forever or does it have a timeout.

Reply:
Agreed that we should set a timeout, even to something large like 30 minutes.

Reply:
sounds reasonable.
I updated the code to include a 30 minutes of timeout.